app/models/notification.rb

```diff
@@ -0,0 +1,55 @@
+class Notification < ActiveRecord::Base
+  belongs_to :activity, inverse_of: :notifications, dependent: :destroy
+
+  acts_as_readable on: :updated_at
+
+  def self.notify(user, course, object, options = {})
```

Comment:
Also, what if I do not want an email notification?

What are all the options you support? You did not document.